DigitalLearningSolutions.Web/Scripts/trackingSystem/reports.ts

```diff
@@ -71,24 +75,37 @@ request.open('GET', path, true);
 request.responseType = 'json';
 request.send();
 
-let allRowsShown = false;
-const viewMoreLink = <HTMLLIElement>document.getElementsByClassName('js-shown-load-more').item(0);
-viewMoreLink.style.display = 'block';
-
 viewMoreLink.addEventListener('click', (event) => {
   event.preventDefault();
-  allRowsShown = !allRowsShown;
-  showMoreRows(allRowsShown);
-  viewMoreLink.innerText = allRowsShown ? 'View Less' : 'View More';
+  const activityRow = <HTMLElement>document.getElementsByClassName(activityToggleRowClass).item(0);
+
+  if (activityRow?.style.display === activityToggleableRowDisplayNone) {
+    viewMoreRows();
+    viewMoreLink.innerText = 'View Less';
+  } else {
+    viewLessRows();
+    viewMoreLink.innerText = 'View More';
+  }
 });
 
-function showMoreRows(status: boolean): void {
+function viewMoreRows(): void {

```

Comment:
These functions look great.
It would be nice to have them somewhere else though - there are still bits of page-load-time code interleaved with function definitions and interfaces and data setup. Once you've created all the functions that are appropriate, can you move all the remaining code that runs on page-load down to the bottom so it's all in one place?